### PR TITLE
Editor/Page Settings: Fix - Layout not updated at first change.

### PR DESCRIPTION
**Pull request by Nevoss**

### CODE REVIEW

File: assets/dev/js/editor/document/hooks/ui/settings/reload-preview.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	}
+
+	apply() {
+		// Logic moved from js assets/dev/js/editor/components/settings/page/manager.js
+		return $e.run( 'document/save/auto', { force: true } )
+			.then( () => {
+				elementor.reloadPreview();
+
+				elementor.once( 'preview:loaded', function() {
+					// The command 'attach-preview' called after the event 'preview:loaded', for that reason
```
Comment (matipojo) [inline assets/dev/js/editor/document/hooks/ui/settings/reload-preview.js:27]: `and it set the default route`

**Review by matipojo — CHANGES_REQUESTED** [review]

File: assets/dev/js/editor/components/settings/page/manager.js [hunk 2]
```diff
@@ -21,19 +21,6 @@ module.exports = BaseSettings.extend( {
 
 			$title.text( newValue );
```
Comment (matipojo) [inline assets/dev/js/editor/components/settings/page/manager.js:22]: Can you check it the `post_title` callback needs also be moved to an hook?

**Review by matipojo — CHANGES_REQUESTED** [review]

**Review by matipojo — DISMISSED** [review]

File: assets/dev/js/editor/document/hooks/ui/settings/reload-preview.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+	getContainerType() {
+		return 'document';
+	}
+
+	getConditions( args ) {
+		return !! args.settings.template;
+	}
+
+	apply() {
+		// Logic moved from js assets/dev/js/editor/components/settings/page/manager.js
```
Comment (kobizz) [inline assets/dev/js/editor/document/hooks/ui/settings/reload-preview.js:21]: Use this comment as part of the commit, not as an inline comment

File: assets/dev/js/editor/document/hooks/ui/settings/reload-preview.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+		return $e.run( 'document/save/auto', { force: true } )
+			.then( () => {
+				elementor.reloadPreview();
+
+				elementor.once( 'preview:loaded', () => {
+					// The command 'attach-preview' that set the default route called after the event 'preview:loaded',
+					// for that reason the function wait 50 ms and then sets page-settings route.
+					setTimeout( () => {
+						$e.route( 'panel/page-settings/settings' );
+					}, 50 );
```
Comment (kobizz) [inline assets/dev/js/editor/document/hooks/ui/settings/reload-preview.js:31]: This is not a good practice. We should find another way to save the previous state.

**Review by kobizz — CHANGES_REQUESTED** [review]

File: assets/dev/js/editor/document/hooks/ui/settings/reload-preview.js (hunk 3 above)
Comment (matipojo) [reply assets/dev/js/editor/document/hooks/ui/settings/reload-preview.js:21]: Why?

File: assets/dev/js/editor/document/hooks/ui/settings/reload-preview.js (hunk 4 above)
Comment (matipojo) [reply assets/dev/js/editor/document/hooks/ui/settings/reload-preview.js:31]: Suggestion?

File: assets/dev/js/editor/document/hooks/ui/settings/reload-preview.js (hunk 3 above)
Comment (matipojo) [reply assets/dev/js/editor/document/hooks/ui/settings/reload-preview.js:21]: @Nevoss move to the commit description.

**Review by matipojo — DISMISSED** [review]

File: assets/dev/js/editor/document/hooks/ui/settings/reload-preview.js (hunk 4 above)
Comment (kobizz) [reply assets/dev/js/editor/document/hooks/ui/settings/reload-preview.js:30]: You can either move the `document:loaded` event to the end of the `attach-preview` command or, alternatively, save the panel state to a variable and use it once the preview is loaded.

**Review by kobizz — CHANGES_REQUESTED** [review]

File: assets/dev/js/editor/document/hooks/ui/settings/reload-preview.js (hunk 4 above)
Comment (kobizz) [reply assets/dev/js/editor/document/hooks/ui/settings/reload-preview.js:30]: 1. Try to move the `document:loaded` event to the end of `attach-preview` command.
2. You can save a state what the panel is open and get back there when the preview is reloaded

**Comment by Aviad-Herman** [comment]
Ref: #12942

**Review by matipojo — CHANGES_REQUESTED** [review]

**Review by matipojo — APPROVED** [review]

**Review by matipojo — CHANGES_REQUESTED** [review]
Fix conflicts

**Review by matipojo — APPROVED** [review]

**Review by kobizz — APPROVED** [review]

**Pull request merged**